packages/store/addon/-private/system/fetch-manager.ts

```diff
@@ -0,0 +1,521 @@
+import { default as RSVP, Promise } from 'rsvp';
+import { DEBUG } from '@glimmer/env';
+import { run as emberRunLoop } from '@ember/runloop';
+import { assert, warn, inspect } from '@ember/debug';
+import Snapshot from './snapshot';
+import { guardDestroyedStore, _guard, _bind, _objectIsAlive } from './store/common';
+import { normalizeResponseHelper } from './store/serializer-response';
+import { serializerForAdapter } from './store/serializers';
+import { InvalidError } from '@ember-data/adapter/error';
+import coerceId from './coerce-id';
+import Ember from 'ember';
+
+import { _findHasMany, _findBelongsTo, _findAll, _query, _queryRecord } from './store/finders';
+import RequestCache from './request-cache';
+import { CollectionResourceDocument, SingleResourceDocument } from '../ts-interfaces/ember-data-json-api';
+import { RecordIdentifier } from '../ts-interfaces/identifier';
+import Store from './store';
+
+function payloadIsNotBlank(adapterPayload): boolean {
+  if (Array.isArray(adapterPayload)) {
+    return true;
+  } else {
+    return Object.keys(adapterPayload || {}).length !== 0;
+  }
+}
+
+const emberRun = emberRunLoop.backburner;
+
+export interface PendingFetchItem {
```

Comment:
move these interfaces to ts-interfaces, especially if eventually made public

Reply:
this are internal to the file, so removed the export and kept them here